sys/geom/geom_subr.c

```diff
@@ -404,6 +396,29 @@ g_new_geomf(struct g_class *mp, const char *fmt, ...)
 	return (gp);
 }
 
+struct g_geom *
+g_new_geom(struct g_class *mp, const char *name)
+{
+	return _g_new_geom(mp, name, strlen(name));
+}
+
+struct g_geom *
+g_new_geomf(struct g_class *mp, const char *fmt, ...)
+{
+	struct g_geom *gp;
+	va_list ap;
+	struct sbuf *sb;
+
+	sb = sbuf_new_auto();
+	va_start(ap, fmt);
+	sbuf_vprintf(sb, fmt, ap);
+	va_end(ap);
+	sbuf_finish(sb);
+	gp = _g_new_geom(mp, sbuf_data(sb), sbuf_len(sb));
```

Comment:
I would just call `g_new_geom(mp, sbuf_data(sb))` here.  Avoiding the length check isn't that big of a deal IMO vs having simpler to read code.